packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,110 +208,265 @@ class CupertinoAlertDialog extends StatelessWidget {
+              child: new _CupertinoDialogRenderWidget(
+                children: <Widget>[
+                  new BaseLayoutId<_CupertinoDialogRenderWidget, MultiChildLayoutParentData>(
+                    id: _AlertDialogSections.actionsSection,
+                    child: _buildActions(),
+                  ),
+                  new BaseLayoutId<_CupertinoDialogRenderWidget, MultiChildLayoutParentData>(
+                    id: _AlertDialogSections.contentSection,
+                    child: _buildContent(),
+                  ),
+                ],
+              ),
+            ),
+          ),
         ),
       ),
     );
   }
 }
 
-/// A button typically used in a [CupertinoAlertDialog].
-///
-/// See also:
-///
-///  * [CupertinoAlertDialog], a dialog that informs the user about situations
-///    that require acknowledgement
-class CupertinoDialogAction extends StatelessWidget {
-  /// Creates an action for an iOS-style dialog.
-  const CupertinoDialogAction({
-    this.onPressed,
-    this.isDefaultAction = false,
-    this.isDestructiveAction = false,
-    @required this.child,
-  }) : assert(child != null);
+// iOS style layout policy widget for sizing an alert dialog's content section and
+// action button section.
+//
+// See [_RenderCupertinoDialog] for specific layout policy details.
+class _CupertinoDialogRenderWidget extends MultiChildRenderObjectWidget {
+  _CupertinoDialogRenderWidget({
+    Key key,
+    @required List<Widget> children,
+  }) : super(key: key, children: children);
 
-  /// The callback that is called when the button is tapped or otherwise
-  /// activated.
-  ///
-  /// If this is set to null, the button will be disabled.
-  final VoidCallback onPressed;
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderCupertinoDialog();
+  }
 
-  /// Set to true if button is the default choice in the dialog.
-  ///
-  /// Default buttons are bold.
-  final bool isDefaultAction;
+  @override
+  void updateRenderObject(BuildContext context, _RenderCupertinoDialog renderObject) {
+    // NO-OP
+  }
+}
 
-  /// Whether this action destroys an object.
-  ///
-  /// For example, an action that deletes an email is destructive.
-  final bool isDestructiveAction;
+// iOS style layout policy for sizing an alert dialog's content section and action
+// button section.
+//
+// The policy is as follows:
+//
+// If all content and buttons fit on screen:
+// The content section and action button section are sized intrinsically and centered
+// vertically on screen.
+//
+// If all content and buttons do not fit on screen:
+// A minimum height for the action button section is calculated. The action
+// button section will not be rendered shorter than this minimum.  See
+// [_RenderCupertinoDialogActions] for the minimum height calculation.
+//
+// With the minimum action button section calculated, the content section is
+// laid out as tall as it wants to be, up to the point that it hits the
```

Comment:
I was just referring to the specific wording. It's not true that 'the content section is laid out as tall as it wants to be' in that it can't arbitrarily decide to be 1000px tall if it doesn't fit in the `constraints.deflate(new EdgeInsets.only(bottom: minActionsHeight))` constraints.

Reply:
hence the other half of the sentence ", up to the point that it hits the minimum button height at the bottom."

Reply:
The minimum button height at the bottom is half of the story no? The parent still imposes its own `constraints` based on _kEdgePadding etc. I'm debating the specific `as tall as it wants to be` wording.

Reply:
We corrected this verbiage together, offline.